assets/js/admin/admin.js

```diff
@@ -27,40 +55,77 @@ jQuery( document ).ready( function( $ ) {
 			const $this = $( element );
 			const val = $this.val();
 			const $row = $this.parents( '.cmb-row' );
+			let valid = false;
 
-			if ( 'required' === $this.data( 'validation' ) ) {
+			if ( 'true' === $this.data( 'required' ) ) {
 				if ( ! val ) {
-					addRequired( $row );
+					addRequiredError( $row );
+					valid = false;
+				} else {
+					valid = true;
+				}
+			}
+
+			if ( $this.data( 'domain' ) && $this.is( ':visible' ) ) {
+				if ( 0 !== val.length && ( ! isUrl( val ) || ! checkUrlDomain( $this.data( 'domain' ), val ) ) ) {
```

Comment:
@jhung Add an additional check to make sure that the input is a URL before validating the domain. Let me know if this works for you!

Reply:
Thanks @greatislander I checked it and it works.